internal/daemon/worker/controller_connection_test.go

```diff
@@ -101,8 +90,9 @@ func TestMonitorUpstreamConnectionState(t *testing.T) {
 				t.Error("Time out waiting for condition")
 			}
 
-			got := upstreamConnectionState.Load()
-			assert.Equal(t, tt.expectedState, got)
+			require.NoError(t, cc.Close())
+			cancelStateCtx()
+			assert.Equal(t, connectivity.Shutdown, cc.GetState())
```

Comment:
Aren't we just testing the gRPC library at this point? Do we need this check?

Reply:
Yeah that's true. I added this this when verifying everything shutdown correctly. I can take this check out. We only care about the states we expect.